src/Components/ProvisioningWizard/steps/Pubkeys/index.js

```diff
@@ -1,19 +1,25 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import { Form, FormGroup, Radio, Text, Title } from '@patternfly/react-core';
-import { useQuery } from '@tanstack/react-query';
+import { useInfiniteQuery } from '@tanstack/react-query';
 import { useWizardContext } from '../../../Common/WizardContext';
 import PubkeySelect from './PubkeySelect';
 import NewSSHKeyForm from './NewKeyForm';
 import { PUBKEYS_QUERY_KEY } from '../../../../API/queryKeys';
 import { fetchPubkeysList } from '../../../../API';
+import { OFFSET } from '../../../../API/helpers';
 
 const EXIST_KEY_OPTION = 'existKey';
 const NEW_KEY_OPTION = 'newKey';
 
 const PublicKeys = ({ setStepValidated }) => {
   const [wizardContext, setWizardContext] = useWizardContext();
-  const { isError, data: pubkeys } = useQuery(PUBKEYS_QUERY_KEY, fetchPubkeysList);
+  const { isError, data } = useInfiniteQuery({
```

Comment:
also when testing, I've discovered, that using `useQuery` and `useInfiniteQuery` under the same key trips the query weirdly, so I wanted to be on the safe side and migrated to usage of infiniteQuery here as well.